sdk/maps/azure-maps-search/README.md

```diff
@@ -92,197 +92,220 @@ Once you initialized a `MapsSearchClient` class, you can explore the methods on
 This library includes a complete async API supported on Python 3.5+. To use it, you must first install an async transport, such as [aiohttp](https://pypi.org/project/aiohttp/).
 See [azure-core documentation](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md#transport) for more information.
 
-Async clients and credentials should be closed when they're no longer needed. These
-objects are async context managers and define async `close` methods.
+Async clients and credentials should be closed when they're no longer needed. These objects are async context managers and define async `close` methods.
 
 ## Examples
 
 The following sections provide several code snippets covering some of the most common Azure Maps Search tasks, including:
 
-- [Request latitude and longitude coordinates for an address](#request-latitude-and-longitude-coordinates-for-an-address)
-
-- [Search for an address or Point of Interest](#search-for-an-address-or-point-of-interest)
-
+- [Geocode an address](#geocode-an-address)
+- [Batch geocode addresses](#batch-geocode-addresses)
+- [Get polygons for a given location](#get-polygons-for-a-given-location)
 - [Make a Reverse Address Search to translate coordinate location to street address](#make-a-reverse-address-search-to-translate-coordinate-location-to-street-address)
-- [Translate coordinate location into a human understandable cross street](#translate-coordinate-location-into-a-human-understandable-cross-street)
-- [Get async fuzzy search batch with param and batchid](#get-async-fuzzy-search-batch-with-param-and-batchid)
-- [Fail to get fuzzy search batch sync](#fail-to-get-fuzzy-search-batch-sync)
-- [Search inside Geometry](#search-inside-geometry)
+- [Batch request for reverse geocoding](#batch-request-for-reverse-geocoding)
 
-- [Working with exist library for Search](#working-with-exist-library-for-search)
-
-### Request latitude and longitude coordinates for an address
+### Geocode an address
 
 You can use an authenticated client to convert an address into latitude and longitude coordinates. This process is also called geocoding. In addition to returning the coordinates, the response will also return detailed address properties such as street, postal code, municipality, and country/region information.
 
 ```python
-from azure.maps.search import MapsSearchClient
+import os
 
-search_result = client.search_address("400 Broad, Seattle");
-```
+from azure.core.exceptions import HttpResponseError
 
-### Search for an address or Point of Interest
+subscription_key = os.getenv("AZURE_SUBSCRIPTION_KEY", "your subscription key")
 
-You can use Fuzzy Search to search an address or a point of interest (POI). The following examples demostrate how to search for `pizza` over the scope of a specific country (`France`, in this example).
+def geocode():
+    from azure.core.credentials import AzureKeyCredential
+    from azure.maps.search import AzureMapsSearchClient
```

Comment:
Can we still keep the exposing class name as `MapsSearchClient`? We can use the [customization](https://github.com/Azure/autorest.python/blob/main/docs/customizations.md) in `_patch.py` to achieve this.